### PR TITLE
Finishing Switch to MekHQ.getLogger() instead of System.out/err/printStackTrace

### PR DESCRIPTION
**Pull request by Windchild292**
This should fix the logger not working in current nightly builds, and completely moves MekHQ to use DefaultMMLogger instead of System.out, System.err, and e.printStackTrace.

To review:
Hide whitespace differences should be on (a bunch of these files have not been opened recently, and it was an automatic thing by my IDE)
View the first three commits together, as they are the same thing (switching to using the getLogger()). Commit four consists of whitespace changes (mostly relating to tabs) that I discovered when reviewing this story.
Only logical change is commented, as are the main questions.

### CODE REVIEW

File: MekHQ/src/mekhq/MekHQ.java [hunk 1]
```diff
@@ -366,8 +366,12 @@ public static void main(String[] args) {
         System.setProperty("com.apple.mrj.application.apple.menu.about.name","MekHQ");
 
         SwingUtilities.invokeLater(() -> {
-            //redirect output to log file
-            redirectOutput();
+            // TODO : logFileName should probably not be inline
+            String logFileName = "logs" + File.separator + "mekhqlog.txt";
+            // Reset the log file - TODO : Maybe move this to being a call to the logger?
+            MegaMek.resetLogFile(logFileName);
```
Comment (Windchild292) [inline MekHQ/src/mekhq/MekHQ.java]: Should this be added to the logger?

File: MekHQ/src/mekhq/campaign/ResolveScenarioTracker.java [hunk 2]
```diff
@@ -860,15 +858,11 @@ private void loadUnitsAndPilots(File unitFile) throws IOException {
...
-            // Read a Vector from the file.
-            try {
+            try (InputStream listStream = new FileInputStream(unitFile)) {
+                // Read a Vector from the file.
                 parser.parse(listStream);
-                listStream.close();
-            } catch (Exception excep) {
-                excep.printStackTrace(System.err);
-                // throw new IOException("Unable to read from: " +
-                // unitFile.getName());
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/ResolveScenarioTracker.java]: This is the only logical change, I've changed listStream into a try (with resources)

**Comment by Windchild292** [comment]
That resource leak is the logger, and is expected.

File: MekHQ/src/mekhq/MekHQ.java (hunk 1 above)
Comment (sixlettervariables) [reply MekHQ/src/mekhq/MekHQ.java]: I don't know but I'm confused as to why the log file bits have to be invoked on the UI thread.

**Comment by Windchild292** [comment]
There are places we could really improve for logging, but it could be worse (and I think this is a step in the correct direction)

File: MekHQ/src/mekhq/MekHQ.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/MekHQ.java]: Good point, I'll test and see what happens outside the UI thread.

Comment (Windchild292) [reply MekHQ/src/mekhq/MekHQ.java]: I've got no idea why it was in the UI thread, but I've moved it to its over thread and added a reset log file method to the default logger.

**Comment by Windchild292** [comment]
One change, and this now requires https://github.com/MegaMek/mekhq/pull/1554

**Pull request merged**